### PR TITLE
[FLINK-16075][docs-zh] Translate "The Broadcast State Pattern" page into Chinese

### PR DESCRIPTION
**Pull request by Zakelly**
## What is the purpose of the change
Translate "The Broadcast State Pattern" page into Chinese

## Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:
* Dependencies (does it add or upgrade a dependency): (no)
* The public API, i.e., is any changed class annotated with @Public(Evolving): (no)
* The serializers: (no)
* The runtime per-record code paths (performance sensitive): (no)
* Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
* The S3 file system connector: (no)
## Documentation
* Does this pull request introduce a new feature? (no)
* If yes, how is the feature documented? (not applicable)

CC @carp84 @klion26

### CODE REVIEW

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 1]
```diff
@@ -136,66 +118,50 @@ public abstract class KeyedBroadcastProcessFunction<KS, IN1, IN2, OUT> {
 }
 {% endhighlight %}
 
-The first thing to notice is that both functions require the implementation of the `processBroadcastElement()` method 
-for processing elements in the broadcast side and the `processElement()` for elements in the non-broadcasted side. 
-
-The two methods differ in the context they are provided. The non-broadcast side has a `ReadOnlyContext`, while the 
-broadcasted side has a `Context`. 
+需要注意的是 `processBroadcastElement()` 是处理的广播流的元素，而 `processElement()` 处理的是另一个流的元素。两个方法的第二个参数(Context)不同，均有以下方法：
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:121]:
```suggestion
需要注意的是 `processBroadcastElement()` 处理广播流的元素，而 `processElement()` 处理另一个流的元素。两个方法的第二个参数(Context)不同，均有以下方法：
```

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 2]
```diff
@@ -136,66 +118,50 @@ public abstract class KeyedBroadcastProcessFunction<KS, IN1, IN2, OUT> {
...
-above.
-
-The difference lies in the type of access each one gives to the broadcast state. The broadcasted side has 
-**read-write access** to it, while the non-broadcast side has **read-only access** (thus the names). The reason for this
-is that in Flink there is no cross-task communication. So, to guarantee that the contents in the Broadcast State are the
-same across all parallel instances of our operator, we give read-write access only to the broadcast side, which sees the
-same elements across all tasks, and we require the computation on each incoming element on that side to be identical 
-across all tasks. Ignoring this rule would break the consistency guarantees of the state, leading to inconsistent and 
-often difficult to debug results.
+这两个方法的区别就是对于 broadcast state 的访问权限不同。在处理广播流元素这端，是**具有读写权限的**，而对于处理非广播流元素这端是**只读**的。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:130]: “这两个方法的区别在于对 broadcast state 的访问权限不同”

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 3]
```diff
@@ -136,66 +118,50 @@ public abstract class KeyedBroadcastProcessFunction<KS, IN1, IN2, OUT> {
...
- 1. the `ReadOnlyContext` in the `processElement()` method gives access to Flink's underlying timer service, which allows
-  to register event and/or processing time timers. When a timer fires, the `onTimer()` (shown above) is invoked with an 
-  `OnTimerContext` which exposes the same functionality as the `ReadOnlyContext` plus 
-   - the ability to ask if the timer that fired was an event or processing time one and 
-   - to query the key associated with the timer.
- 2. the `Context` in the `processBroadcastElement()` method contains the method 
- `applyToKeyedState(StateDescriptor<S, VS> stateDescriptor, KeyedStateFunction<KS, S> function)`. This allows to 
-  register a `KeyedStateFunction` to be **applied to all states of all keys** associated with the provided `stateDescriptor`. 
+同时，`KeyedBroadcastProcessFunction` 在 Keyed Stream 上工作，所以它提供了一些 `BroadcastProcessFunction` 没有的功能:
+ 1. `processElement()` 的参数 `ReadOnlyContext` 提供了方法能够访问 Flink 的定时器服务，可以注册事件或者处理时间的定时器。当定时器触发时，`onTimer()` 方法会被调用，
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:139]: “可以注册事件或者处理时间的定时器” 这里能否把 "event-time 和 processing-time 单独描述下“ 
当定时器触发时，会调用 `onTimer()` 方法

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 4]
```diff
@@ -136,66 +118,50 @@ public abstract class KeyedBroadcastProcessFunction<KS, IN1, IN2, OUT> {
...
+ 提供了 `OnTimerContext`，它具有 `ReadOnlyContext` 的全部功能，并且提供：
+  - 查询当前触发的是一个事件还是处理时间的定时器
+  - 查询定时器关联的key
+ 2. `processBroadcastElement()` 方法中的参数 `Context` 会提供方法 `applyToKeyedState(StateDescriptor<S, VS> stateDescriptor, KeyedStateFunction<KS, S> function)`。
+ 这个方法使用一个 `KeyedStateFunction` 能够对 `stateDescriptor` 对应的 state 的**所有key所有存储的状态**进行某些操作。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:144]:
```suggestion
 这个方法使用一个 `KeyedStateFunction` 能够对 `stateDescriptor` 对应的 state 的**所有 key 的存储状态**进行某些操作。
```

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 5]
```diff
@@ -136,66 +118,50 @@ public abstract class KeyedBroadcastProcessFunction<KS, IN1, IN2, OUT> {
...
 
 <div class="alert alert-info">
-  <strong>Attention:</strong> Registering timers is only possible at `processElement()` of the `KeyedBroadcastProcessFunction`
-  and only there. It is not possible in the `processBroadcastElement()` method, as there is no key associated to the 
-  broadcasted elements.
+  <strong>注意：</strong>注册一个定时器只能在 `KeyedBroadcastProcessFunction` 的 `processElement()` 方法中进行。
+  在 `processBroadcastElement()` 方法中不能注册定时器，因为广播的元素中并没有关联的key。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:148]:
```suggestion
  在 `processBroadcastElement()` 方法中不能注册定时器，因为广播的元素中并没有关联的 key。
```

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 6]
```diff
@@ -249,31 +215,20 @@ new KeyedBroadcastProcessFunction<Color, Item, Rule, String>() {
...
 
-After describing the offered APIs, this section focuses on the important things to keep in mind when using broadcast 
-state. These are:
+这里有一些 broadcast state 的重要注意事项，在使用它时需要时刻清楚：
 
-  - **There is no cross-task communication:** As stated earlier, this is the reason why only the broadcast side of a 
-`(Keyed)-BroadcastProcessFunction` can modify the contents of the broadcast state. In addition, the user has to make 
-sure that all tasks modify the contents of the broadcast state in the same way for each incoming element. Otherwise,
-different tasks might have different contents, leading to inconsistent results.
+  - **没有跨 task 通讯：**如上所述，这就是为什么在 `(Keyed)-BroadcastProcessFunction` 中处理广播流元素的方法里可以更改 broadcast state 的内容。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:222]: 这里要描述出 "only" 也就是只有 broadcast stream 可以修改，非 broadcast stream 不可以修改 broadcast state

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 7]
```diff
@@ -249,31 +215,20 @@ new KeyedBroadcastProcessFunction<Color, Item, Rule, String>() {
...
+  同时，用户需要保证所有 task 对于 broadcast state 的处理方式是一致的，否则会造成不同 task 读取 broadcast state 时内容不一致的情况，最终导致结果不一致。
 
-  - **Order of events in Broadcast State may differ across tasks:** Although broadcasting the elements of a stream 
-guarantees that all elements will (eventually) go to all downstream tasks, elements may arrive in a different order 
-to each task. So the state updates for each incoming element *MUST NOT depend on the ordering* of the incoming
-events.
+  - **broadcast state 在不同的 task 的事件顺序是不同的：**虽然广播流中元素的过程能够保证所有的下游 task 全部能够收到，但在不同 task 中元素的到达顺序可能不同。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:225]: 顺序是不同的 -> "顺序可能是不同的"

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 8]
```diff
@@ -249,31 +215,20 @@ new KeyedBroadcastProcessFunction<Color, Item, Rule, String>() {
...
-when a checkpoint takes place (checkpoint barriers do not overpass elements), all tasks checkpoint their broadcast state, 
-and not just one of them. This is a design decision to avoid having all tasks read from the same file during a restore 
-(thus avoiding hotspots), although it comes at the expense of increasing the size of the checkpointed state by a factor 
-of p (= parallelism). Flink guarantees that upon restoring/rescaling there will be **no duplicates** and **no missing data**. 
-In case of recovery with the same or smaller parallelism, each task reads its checkpointed state. Upon scaling up, each
-task reads its own state, and the remaining tasks (`p_new`-`p_old`) read checkpoints of previous tasks in a round-robin
-manner.
+  - **所有的 task 均会对 broadcast state 进行 checkpoint：**虽然所有 task 中的 broadcast state 是一致的，但当 checkpoint 来临时所有 task 均会对 broadcast state 做 checkpoint。
+  这个设计是为了防止在作业恢复后读文件造成的文件热点。当然这种方式会造成 checkpoint 一定程度的写放大，放大倍数为 p (=并行度)。Flink 会保证在恢复状态/改变并发的时候数据**没有重复**且**没有缺失**。
+  在作业恢复时，如果与之前具有相同或更小的并发度，所有的 task 读取之前已经 checkpoint 过的 state。在并发度上升的情况下，task 会读取本身的 state，多出来的并发 (`p_new` - `p_old`) 会使用轮询调度算法读取之前 task 的 state。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:230]: ”并发度上升“ 改成 ”增大并发“？

File: docs/dev/stream/state/broadcast_state.zh.md [hunk 9]
```diff
@@ -249,31 +215,20 @@ new KeyedBroadcastProcessFunction<Color, Item, Rule, String>() {
...
 
-  - **No RocksDB state backend:** Broadcast state is kept in-memory at runtime and memory provisioning should be done 
-accordingly. This holds for all operator states.
+  - **不使用 RocksDB state backend：** broadcast state 在运行时保存在内存中，需要保证内存充足。这一特性同样适用于 Operator State。
```
Comment (klion26) [inline docs/dev/stream/state/broadcast_state.zh.md:232]: 这里是说对所有的 ”Operator State” 都这样， Broadcast state 是一种特殊的 operator state

**Comment by Zakelly** [comment]
> @Zakelly thanks for your contribution, I left some comments.

Thanks for the detailed review from @klion26 . I have made a new commit.

**Pull request merged**